processor/src/operations/crypto_ops.rs

```diff
@@ -259,22 +253,22 @@ mod tests {
 
         process.execute_op(Operation::MpVerify).unwrap();
         let expected_stack = build_expected(&[
-            leaf[3], leaf[2], leaf[1], leaf[0], depth, index, root[3], root[2], root[1], root[0],
+            node[3], node[2], node[1], node[0], depth, index, root[3], root[2], root[1], root[0],
         ]);
         assert_eq!(expected_stack, process.stack.trace_state());
     }
 
     #[test]
     fn op_mrupdate_copy() {
-        let leaves = init_leaves(&[1, 2, 3, 4, 5, 6, 7, 8]);
+        let nodes = init_nodes(&[1, 2, 3, 4, 5, 6, 7, 8]);
```

Comment:
nit: I'd change the name to just `op_mrupdate`.

Also, I think it is OK to leave the variable here named `leaves` as this is what they actually are.

On a separate note, it might be good to add a test case for updating a non-leaf node and make sure that everything is updated as expected.

Reply:
Added a test that will merge a sub-tree